### PR TITLE
Added edit oasys risk information page

### PR DESCRIPTION
**Pull request by andrewburrowsmoj**
This PR adds an editable oasys risk information page which displays risk information in text boxes. The page does not have any edit functionality in this PR, edit submission is in a future PR. This is okay to merge because changes are behind a feature flag.

commit 4b8054c44c34ca814a49554471049905d5a2e69d

Added POST /referral/:id/confim-edit-oasys-risk-information path.
This links to the view risk information page when the user presses 'Save and continue' after confirming their selection on whether to edit the information or not.

When selected no the user is directed to needs and requirements page.
When selected yes the user is directed to edit OAsys risk information page.

commit ce7f8e29742ec726e166880dd9e3a435cd63c6e9

Refactored view risk information page logic on makeAReferralController.

Split out the logic for rendering a standard risk information page and an oasys risk information page into private methods.
These private methods will be needed when validation fails on confirmation of editing oasys risk information.

commit 6c48c1022e49ae187e4550fc891294ee27a44887

Added edit OASys risk information page.
This page shows the risk information in text area boxes; this commit only contains the page presentation, submission and validation is on a future commit.

commit b22e26c5e61bc93fe7525ca33e4bc1971e359c7d

Split out the OASys risk summary view component into it's own class.
This will be used by both the edit and view OASys risk information pages.

### CODE REVIEW

File: server/routes/makeAReferral/makeAReferralController.ts [hunk 1]
```diff
@@ -526,6 +528,33 @@ export default class MakeAReferralController {
...
+    const referralId = req.params.id
+    const referral = await this.interventionsService.getDraftReferral(accessToken, referralId)
+    const [serviceUser, riskSummary, supplementaryRiskInformation] = await Promise.all([
+      this.communityApiService.getServiceUserByCRN(referral.serviceUser.crn),
+      this.assessRisksAndNeedsService.getRiskSummary(referral.serviceUser.crn, accessToken),
+      this.assessRisksAndNeedsService.getSupplementaryRiskInformationForCrn(referral.serviceUser.crn, accessToken),
+    ])
+
+    if (riskSummary === null) {
+      throw createError(500, `could not obtain risk summary`, {
```
Comment (Gweaton) [inline server/routes/makeAReferral/makeAReferralController.ts:584]: When would Risk summary be `null`? Is it an error from the API or when nothing is found? Just wondering if this should return a 404 (or something to that effect) if the risk summary wasn't found for a specific user

File: server/routes/makeAReferral/makeAReferralController.ts [hunk 2]
```diff
@@ -496,16 +497,37 @@ export default class MakeAReferralController {
...
     const referral = await this.interventionsService.getDraftReferral(res.locals.user.token.accessToken, req.params.id)
-    const [serviceUser, riskSummary] = await Promise.all([
-      this.communityApiService.getServiceUserByCRN(referral.serviceUser.crn),
-      this.assessRisksAndNeedsService.getRiskSummary(referral.serviceUser.crn, res.locals.user.token.accessToken),
-    ])
+    const serviceUser = await this.communityApiService.getServiceUserByCRN(referral.serviceUser.crn)
+    if (config.apis.assessRisksAndNeedsApi.riskSummaryEnabled) {
+      await this.displayOasysRiskInformationPage(res, referral, serviceUser)
+    } else {
+      await this.displayRiskInformationPage(res, referral, serviceUser)
```
Comment (Gweaton) [inline server/routes/makeAReferral/makeAReferralController.ts:505]: Again, could we possibly distinguish these two pages a bit more clearly in their names? `OasysRiskInformationPage` vs `RiskInformationPage` might not be that obvious to readers of the code. I suppose this old route will eventually be deprecated, but could we call it something like `displayInterventionsRiskInformationPage` (or something less bad than that, I'm struggling to think of a better representation, sorry 😄 )

File: server/routes/makeAReferral/makeAReferralController.ts [hunk 3]
```diff
@@ -496,16 +497,37 @@ export default class MakeAReferralController {
...
+  }
+
+  private async displayOasysRiskInformationPage(
+    res: Response,
+    referral: DraftReferral,
+    serviceUser: DeliusServiceUser
+  ) {
+    const { accessToken } = res.locals.user.token
+    const riskSummary = await this.assessRisksAndNeedsService.getRiskSummary(referral.serviceUser.crn, accessToken)
+    // TODO: If riskSummary === null show standard risk information flow or show empty oasys risk information?
```
Comment (Gweaton) [inline server/routes/makeAReferral/makeAReferralController.ts:523]: I'm not sure about this - I'd lean towards showing the standard risk information flow, but is there someone we can ask about this for a second opinion (possibly Leigh?) and capture the response in a Trello ticket so we make sure to revisit it soon?

**Review by Gweaton — APPROVED** [review]
Just had a couple of comments, looks good to me so far though!

File: server/routes/makeAReferral/risk-information/oasys/view/oasysRiskInformationView.ts [hunk 4]
```diff
@@ -1,102 +1,16 @@
...
-    switch (risk.current) {
-      case 'YES':
-        return 'app-oasys-text app-oasys-text--green'
-      case 'NO':
-        return 'app-oasys-text app-oasys-text--red'
-      default:
-        return 'app-oasys-text app-oasys-text--dark-grey'
-    }
+  constructor(readonly presenter: OasysRiskInformationPresenter) {
+    this.riskView = new RiskView(this.presenter.riskPresenter, 'probation-practitioner')
```
[the remark this answers is not in the document]
Comment (andrewburrowsmoj) [reply server/routes/makeAReferral/risk-information/oasys/view/oasysRiskInformationView.ts:13]: yeah, I agree and was what I thought.
Do you want me to add a commit changing the vairable name in this PR?

File: server/routes/makeAReferral/makeAReferralController.ts (hunk 1 above)
Comment (andrewburrowsmoj) [reply server/routes/makeAReferral/makeAReferralController.ts]: ah, I messed up here ... I'm displaying empty risk info if it is null now

File: server/routes/makeAReferral/makeAReferralController.ts (hunk 3 above)
Comment (andrewburrowsmoj) [reply server/routes/makeAReferral/makeAReferralController.ts]: I've removed it now. when it is null it simply provides empty information

File: server/routes/makeAReferral/risk-information/oasys/view/oasysRiskInformationView.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (Gweaton) [reply server/routes/makeAReferral/risk-information/oasys/view/oasysRiskInformationView.ts]: Up to you, but would be good to remember this time, as I forgot... Could add a quick trello card to do it separately, or if you have time now, could tack it onto this PR

**Pull request merged**